File: src/components/AaliyahEngine.cs
public class AaliyahResponse 
{
    public string EarWhisper { get; set; }  // What you hear in your ear
    public string DashboardAlert { get; set; } // What flashes on the screen
    public string SoapCategory { get; set; } // Where it goes in the final note
}

public AaliyahResponse ProcessLiveAudio(string text)
{
    text = text.ToLower();

    // ANOMALY: Verbal Peace vs. Physical Guarding
    if ((text.Contains("fine") || text.Contains("no stress")) && (text.Contains("knot") || text.Contains("trigger point"))) {
        return new AaliyahResponse {
            EarWhisper = "Anomaly: Physical guarding detected. Check for hidden stressors.",
            DashboardAlert = "ANOMALY: Subjective/Objective Mismatch",
            SoapCategory = "Assessment"
        };
        // Add this to your existing analyzeSessionLive logic
if (transcript.includes("used the technique") || transcript.includes("not as painful") || transcript.includes("slept better")) {
    return {
        type: 'Success',
        message: "Clinical Progress Identified: Patient showing improved somatic awareness.",
        action: "Log as 'Treatment Success' for insurance outcome tracking."
    };
}
    }

    // SUCCESS STORY: Integrated Coping
    if (text.Contains("breathed through") || text.Contains("felt better")) {
        return new AaliyahResponse {
            EarWhisper = "Success Marker. Reinforce this behavior.",
            DashboardAlert = "SUCCESS: Somatic Regulation Achieved",
            SoapCategory = "Plan/Assessment"
        };
    }

    // NMT SPECIFIC: Muscle Referral
    if (text.Contains("headache") || text.Contains("temple pain")) {
        return new AaliyahResponse {
            EarWhisper = "Check SCM and Upper Traps for referral patterns.",
            DashboardAlert = "NMT Hint: Active Trigger Point Likely",
            SoapCategory = "Objective"
        };
    }

    return null;
}

File: c-bot/src/Logic/AaliyahEngine.ts
// Define the structure for the Assistant's response
export interface ClinicalInsight {
    type: 'Anomaly' | 'Success' | 'NMT';
    message: string;
    action: string;
}

export const analyzeSessionLive = (text: string): ClinicalInsight | null => {
    const transcript = text.toLowerCase();

    // 1. ANOMALY: Verbal vs. Physical Guarding
    if (transcript.includes("fine") && (transcript.includes("knot") || transcript.includes("tight"))) {
        return {
            type: 'Anomaly',
            message: "Anomaly detected: Physical guarding contradicts verbal report.",
            action: "Probe for hidden social stressors."
        };
    }

    // 2. SUCCESS CASE: Progress Marker
    if (transcript.includes("better") || transcript.includes("managed to")) {
        return {
            type: 'Success',
            message: "Success marker identified.",
            action: "Reinforce somatic self-regulation behavior."
        };
    }

    // 3. NMT HINT: Referral Patterns
    if (transcript.includes("headache") && transcript.includes("temple")) {
        return {
            type: 'NMT',
            message: "Clinical Hint: Active referral pattern detected.",
            action: "Check SCM and Suboccipitals for trigger points."
        };
    }

    return null;
};